invenio_users_resources/services/schemas.py

```diff
@@ -124,3 +168,111 @@ class NotificationPreferences(Schema):
     """Schema for notification preferences."""
 
     enabled = fields.Bool()
+
+
+class DomainOrgSchema(Schema):
+    """Schema for domain orgs."""
+
+    id = fields.Integer(dump_only=True)
+    pid = fields.String(validate=validate.Length(min=1, max=255), required=True)
+    name = fields.String(validate=validate.Length(min=1, max=255), required=True)
+    props = fields.Dict(
+        keys=fields.String(required=True),
+        values=fields.String(validate=validate.Length(max=255)),
+    )
+    is_parent = fields.Boolean(dump_only=True, dump_default=False)
+
+    @validates_schema
+    def validate_props(self, data, **kwargs):
+        """Apply instance specific validation on props."""
```

Comment:
We allow each instance to define their own schema for extra properties on a domain (by default just stores the ``country``).